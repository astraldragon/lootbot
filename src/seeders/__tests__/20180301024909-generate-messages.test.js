jest.mock("../../models", () => {
  return {
    Tier: { findOne: jest.fn("findOne") }
  };
});

const { Tier } = require("../../models");
const migration = require("../20180301024909-generate-messages");

describe("seeders/20180301024909-generate-messages", () => {
  beforeEach(() => {
    jest.spyOn(Date, "now").mockImplementation(() => 1487076708000);
    process.env.GUILD = "1";
    this.guildId = process.env.GUILD;

    this.queryInterface = {
      bulkInsert: jest.fn("createTable").mockReturnValue(true),
      bulkDelete: jest.fn("dropTable").mockReturnValue(true)
    };

    this.Sequelize = {
      Op: { or: "or" }
    };

    Tier.findOne
      .mockReturnValueOnce({ id: 1 })
      .mockReturnValueOnce({ id: 2 })
      .mockReturnValueOnce({ id: 3 })
      .mockReturnValueOnce({ id: 4 });
  });

  afterEach(() => {
    delete process.env.GUILD;
  });

  it("should throw error when seeding if no guild is provided", async () => {
    delete process.env.GUILD;

    await expect(migration.up(this.queryInterface)).rejects.toThrow(
      "GUILD env variable required"
    );
  });

  it("should fetch all the tiers for guild when seeding", async () => {
    await migration.up(this.queryInterface);

    expect(Tier.findOne).toHaveBeenCalledWith({
      where: { guild: this.guildId, name: "Common" }
    });

    expect(Tier.findOne).toHaveBeenCalledWith({
      where: { guild: this.guildId, name: "Uncommon" }
    });

    expect(Tier.findOne).toHaveBeenCalledWith({
      where: { guild: this.guildId, name: "Rare" }
    });

    expect(Tier.findOne).toHaveBeenCalledWith({
      where: { guild: this.guildId, name: "Legendary" }
    });
  });

  it("should seed when guild is provided", async () => {
    await migration.up(this.queryInterface);
    expect(this.queryInterface.bulkInsert).toMatchSnapshot();
  });

  it("should throw error when rolling back if no guild is provided", () => {
    delete process.env.GUILD;

    expect(() => {
      migration.down(this.queryInterface);
    }).toThrowError("GUILD env variable required");
  });

  it("should seed when a guild is provided", () => {
    migration.down(this.queryInterface, this.Sequelize);
    expect(this.queryInterface.bulkDelete).toMatchSnapshot();
  });
});
