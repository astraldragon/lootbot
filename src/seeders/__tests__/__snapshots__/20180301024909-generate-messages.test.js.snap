// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seeders/20180301024909-generate-messages should seed when a guild is provided 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Messages",
      Object {
        "guild": "1",
        "name": Object {
          "or": Array [
            "intro1",
            "intro2",
            "intro3",
            "intro4",
            "intro5",
            "intro6",
            "common1",
            "uncommon1",
            "rare1",
            "legendary1",
            "reward1",
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": true,
    },
  ],
}
`;

exports[`seeders/20180301024909-generate-messages should seed when guild is provided 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Messages",
      Array [
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 1000,
          "guild": "1",
          "message": "Plucky Bookwyrm <user> has opened an MMO Bookclub Lootbox! Let's see what's inside...",
          "name": "intro1",
          "type": "intro",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 1000,
          "guild": "1",
          "message": "<user>, my MMO Bookclub Lootboxes are GUARANTEED to give you a sense of pride and accomplishment. Let's see what you've won...",
          "name": "intro2",
          "type": "intro",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 1000,
          "guild": "1",
          "message": "Knowing that one day they might win a Legendary prize from a MMO Bookclub Lootbox...it fills <user> with determination.",
          "name": "intro3",
          "type": "intro",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 1000,
          "guild": "1",
          "message": "I used to be a bookwyrm like you <user>. And then I took a lootbox to the knee.",
          "name": "intro4",
          "type": "intro",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 1000,
          "guild": "1",
          "message": "<user> used a bookmark to open an MMO Bookclub Lootbox...it's super effective!",
          "name": "intro5",
          "type": "intro",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 1000,
          "guild": "1",
          "message": "<user>, triangulating...",
          "name": "intro6",
          "type": "intro",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 2000,
          "guild": "1",
          "message": "Item, get! You've won a <tier> prize!",
          "name": "common1",
          "tier_id": 1,
          "type": "draw",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 2000,
          "guild": "1",
          "message": "It's dangerous to go alone, take this <user>! You've won an <tier> prize!",
          "name": "uncommon1",
          "tier_id": 2,
          "type": "draw",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 8000,
          "guild": "1",
          "message": "Oh my, it looks like you've won a <tier> prize. I will now pause for eight seconds to build some unbearable tension before the big reveal...",
          "name": "rare1",
          "tier_id": 3,
          "type": "draw",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 9000,
          "guild": "1",
          "message": "Holy smokes, you've only gone and won a frickin' Legendary prize! I will now pause for an excruciating NINE SECONDS to build tension before the big reveal...",
          "name": "legendary1",
          "tier_id": 4,
          "type": "draw",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
        Object {
          "createdAt": 2017-02-14T12:51:48.000Z,
          "delay": 0,
          "guild": "1",
          "message": "Congratulations <user>, you won <reward>! One of your mod overlords will be along shortly to hook you up.",
          "name": "reward1",
          "type": "reward",
          "updatedAt": 2017-02-14T12:51:48.000Z,
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": true,
    },
  ],
}
`;
